lib/queue.rb

```diff
@@ -1,30 +1,64 @@
 class Queue
-
+  
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    # how do we know what the size of the queue is going to be?
+    @store = Array.new() 
+    
+    # this has an index of -1 to indicate these are empty
+    @front = -1
+    @back = -1
+    
   end
-
-  def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+  
+  def enqueue(value)
+    if @front == -1 && @back == -1
+      @store.push(value)
+      @front = 0
+      @back = 0
+      # this adds a viable index number, because for this part of the "if" statement, the array is empty, both front and back are 0
+    else
+      @store.push(value)
```

Comment:
Using push will always put an element to the end of the array, there's no way to "wrap" the back around the internal array.